frigate/events/audio.py

```diff
@@ -168,9 +174,21 @@ def detect_audio(self, audio) -> None:
         if not self.feature_metrics[self.config.name]["audio_enabled"].value:
             return
 
-        waveform = (audio / AUDIO_MAX_BIT_RANGE).astype(np.float32)
+        audio_as_float = audio.astype(np.float32)
+        waveform = audio_as_float / AUDIO_MAX_BIT_RANGE
         model_detections = self.detector.detect(waveform)
 
+        # Calculate RMS (Root-Mean-Square) which represents the average signal amplitude
+        # Note: np.float32 isn't serializable, we must use np.float64 to publish the message
+        rms = np.sqrt(np.mean(np.absolute(audio_as_float**2)))
+
+        # Transform RMS to dBFS (decibels relative to full scale)
+        dBFS = 20 * np.log10(np.abs(rms) / AUDIO_MAX_BIT_RANGE)
+
+        self.stream_metadata_communicator.queue.put(
+            (self.config.name, {"dBFS": float(dBFS), "rms": float(rms)})
+        )
```

Comment:
Might be good to make this publishing a separate function that just gets the audio_as_float passed in

Reply:
Makes sense. Fixed